graphs/check_bipartite_graph_all.py

```diff
@@ -0,0 +1,127 @@
+from collections import defaultdict
+from queue import Queue
+
+
+def check_bipartite(graph: dict[int, list[int]]) -> bool:
+    """
+    Check if a graph is Bipartite using Depth-First Search.
+
+    Args:
+        graph: Adjacency list representing the graph.
+
+    Returns:
+        bool: True if no edge connects same set vertices.
+
+    Examples:
+        >>> is_bipartite(defaultdict(list, {0: [1, 2], 1: [0, 3], ...}))
+        False
+        >>> is_bipartite(defaultdict(list, {0: [1, 2], 1: [0, 2], 2: [0, 1]}))
+        True
+    """
+    queue: Queue = Queue()
+    visited = [False] * len(graph)
+    color = [-1] * len(graph)
+
+    def bfs() -> bool:
```

Comment:
```suggestion
    def breadth-first_search() -> bool:
```
as discussed in `CONTRIBUTING.md`.